core/src/main/c/share/dedup.cpp

```diff
@@ -32,7 +32,14 @@
 #include <cassert>
 #include "dedup.h"
 #include "dedup_comparers.h"
+#include "ooo.h"
+
 #define assertm(exp, msg) assert(((void)msg, exp))
+
+constexpr int64_t ERROR_NOT_SORTED = -1;
+constexpr int64_t NO_TIMESTAMP_DUPLICATES = -2;
+constexpr int64_t ERROR_OUT_OF_RANGE = -2;
```

Comment:
To avoid clashing with potential macros (where errors would be difficult to track down):
```cpp
constexpr int64_t error_not_sorted = -1;
constexpr int64_t no_timestamp_duplicates = -2;
constexpr int64_t error_out_of_range = -2;
```